Change getCurrentState() api to isStateActive() in AlicaContext

veerajsk opened the pull request:
getCurrentState() returns only the active state in the deepest alica plan
tree node. Other active states cannot be known using this method. Therefore
change the api to isStateActive() which can check if a given state is active
in the entire current plan tree.

veerajsk committed:
Change getCurrentState() api to isStateActive() in AlicaContext

getCurrentState() returns only the active state in the deepest alica plan
tree node. Other active states cannot be known using this method. Therefore
change the api to isStateActive() which can check if a given state is active
in the entire current plan tree.

abhi1287 commented:
@abolinsky  Please note. Sootball unit tests will have to be changed to use this api. As of now its going in rr-devel and we will be creating new 0.9.0 branch. Once you do the switch, we will delete 1.0.0 branch.

abhi1287 commented on alica_engine/src/engine/AlicaContext.cpp:
```diff
@@ -53,19 +53,28 @@ bool AlicaContext::isValid()
 }
 
-int64_t AlicaContext::getCurrentState() const
+bool isStateActiveHelper(const RunningPlan* rp, int64_t id)
```
Can make it a private member of AlicaContext?

veerajsk replied:
Made isStateActiveHelper() a private static method of AlicaContext()

abhi1287 commented on alica_engine/src/engine/AlicaContext.cpp:
```diff
@@ -53,19 +53,28 @@ bool AlicaContext::isValid()
-    return deepestState->getId();
+bool AlicaContext::isStateActive(int64_t id) const
+{
+    assert(_engine->getStepEngine());
```
A little comment here would help about why this must be true.

veerajsk replied:
Added comment

veerajsk committed:
Make isStateActiveHelper() a static method of AlicaContext

Make isStateActiveHelper() a static method of AlicaContext instead of
a standalone function to improve code structure.
